src/buffer_test.cc

```diff
@@ -124,4 +125,173 @@ TEST_F(BufferTest, SizeMatrixPaddedStd430) {
   EXPECT_EQ(12U * sizeof(int32_t), b.GetSizeInBytes());
 }
 
+// Creates 10 RGBA pixel values, with the blue channels ranging from 0 to 255,
+// and gets the histogram. Checks that the bin for each blue channel value
+// contains 1, as expected.
+TEST_F(BufferTest, GetHistogramForChannelGradient) {
+  TypeParser parser;
+  auto type = parser.Parse("R8G8B8A8_UINT");
+  Format fmt(type.get());
+
+  // Creates 10 RBGA pixel values with the blue channels ranging from 0 to 255.
+  // Every value gets multiplied by 25 to create a gradient
+  std::vector<Value> values(40);
+  for (uint32_t i = 0; i < values.size(); i += 4)
+    values[i + 2].SetIntValue(i / 4 * 25);
+
+  Buffer b(BufferType::kColor);
+  b.SetFormat(&fmt);
+  b.SetData(values);
+
+  std::vector<uint64_t> bins = b.GetHistogramForChannel(2, 256);
+  for (uint32_t i = 0; i < values.size(); i += 4)
+    EXPECT_EQ(1, bins[i / 4 * 25]);
+}
+
+// Creates 10 RGBA pixel values, with all channels being 0, and gets the
+// histogram.
+TEST_F(BufferTest, GetHistogramForChannelAllBlack) {
+  TypeParser parser;
+  auto type = parser.Parse("R8G8B8A8_UINT");
+  Format fmt(type.get());
+
+  std::vector<Value> values(40);
+  for (uint32_t i = 0; i < values.size(); i++)
+    values[i].SetIntValue(0);
+
+  Buffer b(BufferType::kColor);
+  b.SetFormat(&fmt);
+  b.SetData(values);
+
+  for (uint8_t i = 0; i < 4; i++) {
+    std::vector<uint64_t> bins = b.GetHistogramForChannel(i, 256);
+    for (uint32_t y = 0; y < values.size(); y++)
+      EXPECT_EQ(10, bins[0]);
+  }
+}
+
+// Creates 10 RGBA pixel values, with all channels being the maximum value of 8
+// bit uint, and gets the histogram.
```

Comment:
"gets the histogram"

Not a big deal, but since there is typo below that is worth fixing:

I would replace "gets the histogram" with "checks that..." for all of the above tests. E.g. in this case: checks that all channels have a count of 10 (all pixels) in the 255 (max uint8_t) bin.

Reply:
Done. Hopefully it's correct.